src/classes/ALLO_PaymentSync_SVC.cls

```diff
@@ -378,24 +397,43 @@ public class ALLO_PaymentSync_SVC {
 
                 Decimal totalOpportunityAllocations = context.totalOpportunityAllocationsByGAU.get(gau);
 
+                System.debug(loggingLevel.ERROR, 'totalOpportunityAllocations: ' + totalOpportunityAllocations);
+                System.debug(loggingLevel.ERROR, 'newGAUAmtForOpp: ' + newGAUAmtForOpp);
+
                 //check if they need updating
                 if (newGAUAmtForOpp != totalOpportunityAllocations) {
+                    Map<Id, Allocation__c> gauToOpportunityMap = new Map<Id, Allocation__c>();
                     for (Allocation__c oppAllocation : context.opportunityAllocationsByGAU.get(gau)) {
-                        //we want to keep the ratios amongst duplicate opportunity allocations
-                        Decimal duplicateAllocationPaidUnpaidAmountRatio = oppAllocation.Amount__c / totalOpportunityAllocations;
+                        if (!gauToOpportunityMap.containsKey(gau)) {
+                            System.debug(loggingLevel.ERROR, 'oppAllocation is: '  + oppAllocation);
+                            //we want to keep the ratios amongst duplicate opportunity allocations
+                            Decimal duplicateAllocationPaidUnpaidAmountRatio = oppAllocation.Amount__c / totalOpportunityAllocations;
+                            System.debug(loggingLevel.ERROR, 'duplicateAllocationPaidUnpaidAmountRatio: ' + duplicateAllocationPaidUnpaidAmountRatio);
 
-                        oppAllocation.Amount__c = newGAUAmtForOpp * duplicateAllocationpaidUnpaidAmountRatio;
-                        if (oppAllocation.Percent__c != null) {
-                            oppAllocation.Percent__c = (context.Opportunity.Amount != 0) ? oppAllocation.Amount__c / context.opportunity.Amount * 100 : 0;
-                        }
+                            // Not sure why we are multiplying by the duplicateAllocationPaidUnpaidAmountRatio??
+                            // oppAllocation.Amount__c = newGAUAmtForOpp * duplicateAllocationpaidUnpaidAmountRatio;
```

Comment:
@jbrandolinisf  Why are we multiplying by the duplicateAllocationPaidUnpaidAmountRatio?

Reply:
Basically, it is determining what percentage of the pie the GAU is of the total amount for that GAU.  It uses this to preserve the ratios of the allocations on the Opportunity if there are duplicates.  Where there are not duplicates, it is 1.  For example, I have 2 allocs for GAU 1.  One is for 25, and the other is for 75.  

Now, lets say the Allocations on the Payments for GAU 1 are doubled, and now total 200, but the Opportunity is not over or under scheduled.

When it processes the first entry for GAU 1, it will be come 200 * .25 = 50.  When it processes, the second entry in the opportunity allocations for GAU 1, it will be come 200 * .75 = 150.  150 + 50 = 200....

This works for the following reasons:

1) It is calculating the totalAmountAllocatedFor that GAU, before updating it, so it can use it as a ratio with the existing amount.
2) We are determining the total amount for a GAU across the paid and unpaid allocations.

Hope this helps to clarify what it is/should be doing and why.  If you believe it needs to change, we should probably talk it through.  This approach would continue to maintain both allocations on Opportunity, using their original ratios to decide how to split the updated amount.